experimenter/experimenter/experiments/tests/api/v5/test_serializers/test_nimbus_ready_for_review_serializer.py

```diff
@@ -1257,61 +1257,121 @@ def test_no_treatment_branches_for_rollout(self):
             [NimbusConstants.ERROR_SINGLE_BRANCH_FOR_ROLLOUT],
         )
 
-    def test_rollout_valid_version_support(self):
-        desktop = NimbusExperiment.Application.DESKTOP
+    @parameterized.expand(
+        [
+            (
+                NimbusExperiment.Application.DESKTOP,
+                NimbusExperiment.Version.FIREFOX_105,
+                NimbusExperiment.TargetingConfig.MAC_ONLY,
+            ),
+            (
+                NimbusExperiment.Application.IOS,
+                NimbusExperiment.Version.FIREFOX_105,
+                # NimbusExperiment.TargetingConfig.IOS_CORE_ACTIVE_USER,
+            ),
+            (
+                NimbusExperiment.Application.FOCUS_IOS,
+                NimbusExperiment.Version.FIREFOX_105,
+                # NimbusExperiment.TargetingConfig.IOS_CORE_ACTIVE_USER,
+            ),
+            (
+                NimbusExperiment.Application.FENIX,
+                NimbusExperiment.Version.FIREFOX_105,
+                # NimbusExperiment.TargetingConfig.MOBILE_RECENTLY_UPDATED,
+            ),
+            (
+                NimbusExperiment.Application.FOCUS_ANDROID,
+                NimbusExperiment.Version.FIREFOX_105,
+                # NimbusExperiment.TargetingConfig.MOBILE_RECENTLY_UPDATED,
+            ),
+        ]
+    )
+    def test_rollout_valid_version_support(
+        self, application, firefox_version, targeting_config
+    ):
         experiment = NimbusExperimentFactory.create_with_lifecycle(
             NimbusExperimentFactory.Lifecycles.CREATED,
-            application=desktop,
+            application=application,
             channel=NimbusExperiment.Channel.NO_CHANNEL,
-            firefox_min_version=NimbusExperiment.Version.FIREFOX_108,
-            feature_configs=[NimbusFeatureConfigFactory(application=desktop)],
+            firefox_min_version=firefox_version,
+            feature_configs=[NimbusFeatureConfigFactory(application=application)],
             is_sticky=False,
             is_rollout=True,
-            targeting_config_slug=NimbusExperiment.TargetingConfig.MAC_ONLY,
+            targeting_config_slug=targeting_config,
         )
         experiment.save()
         for branch in experiment.treatment_branches:
             branch.delete()
 
         data = {
-            "application": NimbusExperiment.Application.DESKTOP,
+            "application": application,
             "is_sticky": "false",
             "is_rollout": "true",
-            "targeting_config_slug": NimbusExperiment.TargetingConfig.MAC_ONLY,
+            "targeting_config_slug": targeting_config,
             "changelog_message": "test changelog message",
             "channel": "",
```

Comment:
and if the failure is channel missing, then add a channel

Reply:
Thank for the feedback @yashikakhurana Adding the `channel` removed the `channel missing errors` in one the tests, 
Currently, I am getting this error for other Application Versions except for Desktop, 

```
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
experimenter/experiments/tests/api/v5/test_serializers/test_nimbus_ready_for_review_serializer.py:1319: in test_rollout_valid_version_support
    self.assertTrue(serializer.is_valid())
E   AssertionError: False is not true
_ TestNimbusReviewSerializerSingleFeature.test_rollout_valid_version_support_3_ios _
```

what might I be missing?

Reply:
if you are getting this message, I would try to see what serialize.errors are

Reply:
On debugging the tests I found out that for valid rollout, it requires `is_sticky` to be `true`
currettly we have `is_sticky : true`, 

On validating `_validate_sticky_enrollment` we ge this error
![Screenshot 2023-03-28 at 01 41 02](https://user-images.githubusercontent.com/29012135/228083736-295c1f12-01f3-4d5e-948e-1c738bf6efa6.png)

Thank's @yashikakhurana  I have fixed it now, pushing the changes in a few minutes.